x/iro/types/msgs.go

```diff
@@ -0,0 +1,113 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var (
+	_ sdk.Msg = &MsgCreatePlan{}
+	_ sdk.Msg = &MsgBuy{}
+	_ sdk.Msg = &MsgSell{}
+	_ sdk.Msg = &MsgClaim{}
+	_ sdk.Msg = &MsgUpdateParams{}
+)
+
+func (m *MsgCreatePlan) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(m.Owner)
+	if err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("invalid owner address: %s", err)
+	}
+	plan := NewPlan(1, m.RollappId, sdk.NewCoin("dummy", m.AllocatedAmount), m.BondingCurve, m.StartTime, m.PreLaunchTime, m.IncentivePlanParams)
+	return plan.ValidateBasic()
```

Comment:
why do create a dummy plan here and validate it with dummy values vs not just validating the MsgCreatePlan?
for example, in the plan we validate that `ClaimedAmt` and `SoldAmt` are not negative, but in essence in the `MsgCreatePlan` they should be validated against 0.

Reply:
it validates the user's inputs. the dummy plan is just to avoid code dup

Reply:
not sure I got u. 
refactored anyway